qiskit_sphinx_theme/static/css/theme.css

```diff
@@ -34,6 +34,11 @@
   --breakpoint-xl: 1200px;
   --font-family-sans-serif: "IBM PLex Sans", Roboto, "Helvetica Neue", Arial, sans-serif;
   --font-family-monospace: "IBM PLex Mono", Consolas, "Courier New", monospace;
+  --header-height: 52px;
+}
+
+section {
+  scroll-margin-top: var(--header-height);
 }
```

Comment:
Maybe worth adding a comment explaining what this is for, linking to this PR or the relevant issue. Save people from needing to check Git Blame